app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -347,6 +347,10 @@ function paginate(tag, params, requiresApproval) {
   var homeEl = document.getElementById('index-container');
   if (homeEl.dataset.feed === 'base-feed') {
     searchHash.class_name = 'Article';
+    const isHomePageFeed = window.location.pathname == '/';
+    if (isHomePageFeed && userData()) {
```

Comment:
So, just to highlight here, this feels a bit ... fragile? And _still_ ... all the better ways I've thought of to "thread the needle" here (so far) are just much more work. Basically, we want to signal to the back-end to filter out the "antitags", but the backend query models we're using aren't good at distinguishing what "mode" we're in on the front-end. And, on the front-end, because we re-use `base-feed` in so many places, there's not a _lot_ that really distinguishes "home page feed" from all the other "feed-like" pages (`/username`, `/t/tagname`, etc.) 

(And we can't put the "antitags" data as "close" to the fetch as I'd otherwise like due to the caching layers for the home feed making `current_user` async-only.)

I'm certainly open to suggestions if anyone has thoughts on other approaches, but my sense is that this is the best short-term path.

I am a bit worried about how much debt we continue to pile onto the "stories controller", but all signs suggest that we haven't hit the moment for that refactor... _yet_.

Reply:
> However I am a little confused about why we can't just conditionally use current_user (or maybe session_current_user_id?)

We probably _could_, but I think we would still need something to prompt us to _try_. 

I think I explained myself poorly here, so I'll try another attempt:

So: `search_controller#feed` has a method, `is_homepage_search`, that is used to detect any/all homepage "search-like" interactions (which — relevant here — actually _includes_ infinite scrolling). For the purposes of this feature, we need to split hairs and distinguish "homepage feed pagination" from "a search that originated from the article feed", because they will now apply different filters.

Rather than somehow signal to this controller something like "this is _really_ a homepage feed pagination" (in which the controller could then apply the antitags filter using `current_user`), I had the front-end conditionally include `antitags` in the search params. I'm a bit mixed on this approach — in general, I'd much prefer this `SearchController` could get a full refactor, and in specific, it feels like we'd kind of _prefer_ the controller were some kind of `home feed controller` (and not a search mechanism) — but my attempts to do this resulted in _significantly_ more code, and I'm not sure it's worth pursuing as part of this feature.

What I _would_ really _like_ to have done, would have been to locate the `antitags` data-attribute much _closer_ to where it's being used, instead of bringing it in from a distant and removed `async_info` attribute. (User "antitags" feels like a filter we're invisibly applying to an article search, not really so much an aspect of "user data".) However, because of caching, we are again kind of stuck with adding more onto `async_info`, for better or worse.

Does that maybe explain it any better?

I'm still open to alternative implementations, if there's better ways to approach this!